app/models/compute_resources/foreman/model/libvirt.rb

```diff
@@ -278,7 +278,9 @@ def create_volumes(args)
         (volumes = args[:volumes]).each do |vol|
           vol.name = "#{args[:prefix]}-disk#{volumes.index(vol) + 1}"
           vol.capacity = "#{vol.capacity}G" unless vol.capacity.to_s.end_with?('G')
-          vol.allocation = "#{vol.allocation}G" unless vol.allocation.to_s.end_with?('G')
+          if vol.allocation.match(/^\d+/) && !(vol.allocation.to_s.end_with?('G'))
```

Comment:
@xprazak2 This should now solve the problem of `GG`.
For `undefined method [] for nil:NilClass` error, I've tried to make the change in [#fog-libvirt/62](https://github.com/fog/fog-libvirt/pull/62). Can we once again review the PRs together to see if this still leads to any issue?

Reply:
Works well with the fog patch.

Reply:
[#fog-libvirt/62](https://github.com/fog/fog-libvirt/pull/62) has now been merged.

Reply:
Nice work! I asked maintainer for a new release. I would wait with merge until we have the new gem version so that we do not forget to get it into packaging.

Reply:
Sure, makes sense. We'll wait for their new release.

Reply:
@apuntamb Rubocop error :point_up:

Reply:
thanks. done